inorbit_edge/tests/test_robot_session_callbacks.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from unittest.mock import MagicMock
+import pytest
+from inorbit_edge.robot import RobotSession
+from paho.mqtt.client import MQTTMessage
+from inorbit_edge.inorbit_pb2 import Echo
+import time
+from inorbit_edge.inorbit_pb2 import CustomScriptCommandMessage
+
+
+def test_builtin_callbacks(mock_mqtt_client, mock_inorbit_api):
+    robot_session = RobotSession(
+        robot_id="id_123",
+        robot_name="name_123",
+        api_key="apikey_123",
+    )
+    # connect robot_session so it populates properties with API response data
+    robot_session.connect()
+    robot_session._on_connect(..., ..., ..., 0)
+
+    robot_session.client.subscribe.assert_any_call(topic="r/id_123/ros/loc/set_pose")
+    robot_session.client.subscribe.assert_any_call(
+        topic="r/id_123/custom_command/script/command"
+    )
+    robot_session.client.subscribe.call_count == 2
+
+
+def test_robot_session_register_command_callback(mock_mqtt_client, mock_inorbit_api):
+    def my_command_handler(command_name, args, options):
+        pass
+
+    robot_session = RobotSession(
+        robot_id="id_123",
+        robot_name="name_123",
+        api_key="apikey_123",
+    )
+    robot_session.register_command_callback(my_command_handler)
+
+    # connect robot_session so it populates properties with API response data
+    robot_session.connect()
+    # manually execute on_connect callback so the ``custom_command_callback``
+    # callback gets registered
+    robot_session._on_connect(..., ..., ..., 0)
+
+    assert my_command_handler in robot_session.command_callbacks
+    robot_session.client.subscribe.assert_called_with(
+        topic="r/id_123/custom_command/script/command"
+    )
```

Comment:
isn't this already implicitly covered by `test_robot_session_executes_command_callback_on_message` ? I have the non-strong feeling that checking this internal behavior in the tests isn't necessary or a good idea.

Reply:
No, they are different tests. This one only checks that the SDK subscribes to the custom command topic when a callback is registered. `test_robot_session_executes_command_callback_on_message` checks that the callbacks are executed with specific parameters